positional_encodings/positional_encodings.py

```diff
@@ -76,7 +76,40 @@ def forward(self, tensor):
         emb[:, :, : self.channels] = emb_x
         emb[:, :, self.channels : 2 * self.channels] = emb_y
 
-        return emb[None, :, :, :orig_ch].repeat(batch_size, 1, 1, 1)
+        return emb[None, :, :, :orig_ch]
+
+class FixedPositionalEncoding2D(nn.Module):
+    def __init__(self, shape, channels):
+        super(FixedPositionalEncoding2D, self).__init__()
+        self.pos_encoder = SinglePositionalEncoding2D(channels)
+        self.pos_encoding = self.pos_encoder(torch.ones(1, shape[0], shape[1], channels))
+        self.device = self.pos_encoding.device
+        self.repeated_pos_encoding = self.pos_encoding.repeat(32, 1, 1, 1)
```

Comment:
is 32 just arbitrary?

Reply:
We could make it an input parameter probably...

Reply:
32 is indeed out of thin air. I needed an initial batch size and 32 is a very standard batch size so I went for that one. It really doesn't matter at all what you do here because it will be overwritten the first time you call the model. But yeah I agree that adding a key parameter would look nicer.

Reply:
Anyways I think what you did is good :+1: